src/core/mesh/qgsmeshlayer.h

```diff
@@ -377,6 +377,29 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
       */
     void setReferenceTime( const QDateTime &referenceTime );
 
+
+    /**
+      * Returns the exact position in map coordinates of the closest vertex in the search area
+      *
+      * \param point the center of the search area in map coordinates
+      * \param searchRadius the radius of the search area in map units
+      * \return the exact position of the vertex in map coordinates, empty if no edge
+      *
+      * \since QGIS 3.14
+      */
+    QgsPointXY snapOnVertex( const QgsPointXY &point, double searchRadius );
```

Comment:
Can you please create only one function with additional parameter QgsMesh::ElementType similar to https://github.com/qgis/QGIS/blob/768d68f2bf862cbc979970e2c398c6d47dbd9008/src/core/mesh/qgsmeshspatialindex.h#L68 ?

Reply:
it is ok to implement it for faces too

Reply:
> Can you please create only one function with additional parameter QgsMesh::ElementType similar to https://github.com/qgis/QGIS/blob/768d68f2bf862cbc979970e2c398c6d47dbd9008/src/core/mesh/qgsmeshspatialindex.h#L68 ?
I did think at this way too, but as we don't need snap on faces and that snap for faces has no sens for me (see below), I to preferred split the implementation. But I have no problem to make only one function.


> it is ok to implement it for faces too
as the snap is 2D, snapping on faces doesn't make sens for me... But if we implement only one function for all element with QgsMesh::ElementType, it have to be implemented too. Waht do you want the function return for faces ? Directly the entry point ?

Reply:
best to have one function to be consistent with the rest of the API (e.g. spatialindex). Implementation of snapping on faces centroids is 0-work, since the centroids are already calculated in triangular mesh and it does not hurt to have it.... ?